plugins/kubernetes/app/models/kubernetes/job_executor.rb

```diff
@@ -0,0 +1,96 @@
+# Execute a job that can build/push Docker images
+# and write progress to the local job output.
+module Kubernetes
+  class JobExecutor
+    WAIT_FOR_JOB = 5.minutes
+    TICK = 2.seconds
+
+    attr_reader :job
+    delegate :id, to: :job
+
+    def initialize(output, job:)
+      @output = output
+      @job = job
+    end
+
+    def build!(reference, repo:, tag:, push: false, registry:)
+      job_log = ""
```

Comment:
Seems strange to have this write to `@output`, and also to return a string with the job output.  Why not just write all output to `@output`?